src/lib/utils.ts

```diff
@@ -0,0 +1,90 @@
+import { parse as sdlParse } from "graphql/language/parser";
+import { print as sdlPrint } from "graphql/language/printer";
+
+export function headerArrayToObject(
+  headers: [boolean, string, string, boolean][]
+): Record<string, string> {
+  let res = {} as Record<string, any>;
+  for (let header of headers)
+    if (header[0] && header[1].length)
+      res[header[1] as unknown as string] = header[2];
+  return res;
+}
+
+export function headerObjectToArray(
+  headers: Record<string, string>,
+  hidden: string[]
+): [boolean, string, string, boolean][] {
+  return Object.entries(headers).map((e) => [
+    true,
+    ...e,
+    hidden.includes(e[0]),
+  ]);
+}
+
+export function edited2DArray(
+  arr: [boolean, string, string, boolean][],
+  row: number,
+  col: number,
+  val: string | boolean
+): [boolean, string, string, boolean][] {
+  let res = arr.slice();
+  res[row][col] = val;
+  return res;
+}
+
+export const toggleCacheDirective = (
+  operationString: string | undefined
+): string => {
+  if (!operationString) return "";
+  let operationAst;
+  try {
+    operationAst = sdlParse(operationString);
+  } catch (e) {
+    console.error(e);
+    return "ERROR";
+  }
+
+  const shouldAddCacheDirective = !operationAst.definitions.some((def: any) => {
+    return def.directives.some((dir: any) => dir.name.value === "cached");
+  });
+
+  const newOperationAst = JSON.parse(JSON.stringify(operationAst));
+
+  newOperationAst.definitions = operationAst.definitions.map((def: any) => {
+    if (def.kind === "OperationDefinition" && def.operation === "query") {
+      const newDef = {
+        ...def,
+        directives: def.directives.filter(
+          (dir: any) => dir.name.value !== "cached"
+        ),
+      };
+      if (shouldAddCacheDirective) {
+        newDef.directives.push({
+          kind: "Directive",
+          name: {
+            kind: "Name",
+            value: "cached",
+          },
+        });
+      }
+      return newDef;
+    }
+    return def;
+  });
+
+  try {
+    const newString = sdlPrint(newOperationAst);
+    return newString;
+  } catch {
+    throw new Error("cannot build the operation string");
+  }
+};
+
+export function debounce(fn: Function, delay: number) {
```

Comment:
Awesome!

Reply:
> That's probably because graphiql-container is a graphiql specific class and it might not not be trivial to override it.
Would work if we write a separate class and pass the className of this div as graphiql-container custom-class. This doesn't work either?

The actual problem was something unrelated. I had given `height: "100vh"` instead of `height: 100vh`. That worked in inline styles and not in CSS. Fixed now.